tests/tests_unit.c

```diff
@@ -502,6 +502,83 @@ static void test_pbkdf2(void)
                     32);
 }
 
+static void test_create_public_key(void)
```

Comment:
I'd like to work with you on how to facilitate the public key functionality. Perhaps I should create the PR against a different branch?  Or would you rather resolve within this PR?